modules/custom/az_block_types/az_custom_menu_block/src/EventSubscriber/AzCustomMenuBlockKernelViewSubscriber.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\az_custom_menu_block\EventSubscriber;
+
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Alters the block library modal.
+ *
+ * We can't use hook_ajax_render_alter() because the #markup is rendered before
+ * it is passed to that hook.
+ */
+class AzCustomMenuBlockKernelViewSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $currentRouteMatch;
+
+  /**
+   * Constructs a new MenuBlockKernelViewSubscriber.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $current_route_match
+   *   The current route match.
+   */
+  public function __construct(RouteMatchInterface $current_route_match) {
+    $this->currentRouteMatch = $current_route_match;
+  }
+
+  /**
+   * Alters the block library modal.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
```

Comment:
I may be way off on this, but to resolve the PHPStan errors, does this need to reference GetResponseForControllerResultEvent?
- [getControllerResult()](https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/function/GetResponseForControllerResultEvent%3A%3AgetControllerResult/9.3.x)
- [setControllerResult()](https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/function/GetResponseForControllerResultEvent%3A%3AsetControllerResult/9.3.x)

Reply:
Let's try it.

Reply:
Actually, that is deprecated https://api.drupal.org/api/drupal/vendor%21symfony%21http-kernel%21Event%21GetResponseForControllerResultEvent.php/9.3.x